source/slang/hlsl.meta.slang

```diff
@@ -11404,19 +11404,64 @@ ${{{{
 }
 
 //
-// Vulkan/SPIR-V specific features
+// SubpassInput
 //
 
-struct VkSubpassInput<T>
+__magic_type(SubpassInputType)
+__intrinsic_type($(kIROp_SubpassInputType))
+struct __SubpassImpl<T, let isMS:int>
 {
-    T SubpassLoad();
 }
 
-struct VkSubpassInputMS<T>
+__generic<T, let isMS:int=0>
+typealias SubpassInputMS = __SubpassImpl<T, isMS>;
+
+__generic<T, let isMS:int=0>
+extension __SubpassImpl<T, isMS>
 {
-    T SubpassLoad(int sampleIndex);
+    [ForceInline]
+    T SubpassLoad()
+    {
+        __target_switch
+        {
+        case hlsl: __intrinsic_asm "$0.SubpassLoad()";
+        case spirv:
+        {
+            let zeroVec = int2(0);
+            return spirv_asm
+            {
+                OpCapability StorageImageReadWithoutFormat;
+                result:$$T = OpImageRead $this $zeroVec
+            };
+        }
+        }
+    }
 }
 
+__generic<T, let isMS:int=1>
+typealias SubpassInput = __SubpassImpl<T, isMS>;
+
+__generic<T, let isMS:int=1>
+extension __SubpassImpl<T, isMS>
```

Comment:
Again, I don't know if the intention is to provide the function for `isMS=1` only, if so this needs to be changed too.